clojure-mode.el

```diff
@@ -979,18 +979,31 @@ nil."
 
 
 
-(defun clojure-expected-ns ()
-  "Return the namespace name that the file should have."
-  (let* ((project-dir (file-truename
-                       (or (locate-dominating-file default-directory
-                                                   "project.clj")
-                           (locate-dominating-file default-directory
-                                                   "build.boot"))))
-         (relative (substring (file-truename (buffer-file-name))
-                              (length project-dir)
-                              (- (length (file-name-extension (buffer-file-name) t))))))
-    (replace-regexp-in-string
-     "_" "-" (mapconcat 'identity (cdr (split-string relative "/")) "."))))
+(defun clojure-project-dir ()
+  "Return the absolute path to the project's root directory."
+  (file-truename
+   (or (locate-dominating-file default-directory
+                               "project.clj")
+       (locate-dominating-file default-directory
+                               "build.boot"))))
+
+(defun clojure-project-relative-path (path)
+  "Denormalize PATH by making it relative to the project root."
+  (substring path (length (clojure-project-dir))))
```

Comment:
Yeah, this makes sense.  In my defense I didn't write that code, I just moved it out into a utility function :p 

Thanks for the link to the emacs manual, I had no idea that function existed.